packages/web/src/scss/tools/_utilities.scss

```diff
@@ -1,13 +1,56 @@
-// Inspired by React UI and Bootstrap
+// 1. Format key if it's type of number so it's always 2 digits
+// 2. Don't prefix if value key is null (e.g. with shadow class)
+// 3. Remove default postfix from class name
+
+@use '@tokens' as tokens;
 @use 'sass:list';
 @use 'sass:map';
 @use 'sass:meta';
-@use 'string';
+@use 'sass:string';
+@use 'string' as spirit-string;
+@use 'tokens' as tokens-tools;
+
+// Generates a map of tokens based on the provided key, prefix, replacement, and scope.
+// It iterates over the tokens and creates a new map with modified keys and values.
+// Parameters are:
+// * $tokens-key: the key used to retrieve a group of tokens from `@tokens`.
+// * $prefix: a prefix to prepend to each key in the generated map.
+// * $replacement: a string to be replaced in the token keys.
+// * $scope: a list of keys to filter tokens included in the generated map.
+// Returns a map of tokens with modified keys and values.
+@function generate-token-map($tokens-key: null, $prefix: null, $replacement: null, $scope: null) {
```

Comment:
Proposed, up for discussion.

* tokens-group (no default - required) - former tokens-key
* keys-to-generate (no default - required) - former scope
* prefix (default `''`)
* remove-from-name (default `''`) - former replacement

Reply:
Great, sounds much better. Thanks!